### PR TITLE
Rma mode

### PR DESCRIPTION
**Pull request by ycheng627**
Create UI for RMA mode.

### CODE REVIEW

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  >
+    <input
+      type="checkbox"
+      (click)="handleCheckboxClick($event, index)"
+      [disabled]="isRunning || isUnsupported(index)"
+      [checked]="testList[index].enabled"
+    />
+    {{ testConfig.title }}
+    <span class="show-icon">
+      <!-- <mat-icon style="color: red;">cancel</mat-icon> -->
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.html:43]: What is this line for?

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  stopTests() {
+    this.testOrchestrator.stopTests();
+  }
+
+  getCurrentTestIndex$() {
+    return this.testOrchestrator.getCurrentTestIndex$();
+  }
+
+  hasPassed(index: number): 'null' | 'passed' | 'failed' {
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts:61]: How about `getTestResult(index: number): 'pending' | 'passed' | 'failed'`?

File: diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+  @Input() title!: string;
+  @Input() argument!: RoutineV2Argument;
+  @Output() testCompleted = new EventEmitter<void>();
+
+  public RoutineV2State = RoutineV2State;
+
+  // The output, which may be either routine result or user action.
+  public output?: object | string;
+  // Whether the routine has passed. Only shown if the routine is finished.
+  public hasPassed?: boolean;
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts:50]: Why optional? Could we just use `public hasPassed = false`?

File: diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+  ngOnInit() {
+    const supported = this.testOrchestrator.getTestConfig(this.index)
+      ?.supported;
+    if (supported !== null && !supported) {
+      this.output = 'Routine unsupported';
+      return;
+    }
+
+    this.routineV2Service
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts:82]: Use `async ngOnInit` and `try-await-catch`

File: diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        delete routineFinishedInfo.uuid;
+        delete routineFinishedInfo.has_passed;
+        this.output = routineFinishedInfo;
+        this.onTestComplete();
+        break;
+      }
+    }
+  }
+
+  private _checkUUID(uuid: string | undefined): boolean {
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts:180]: Remove the underscore

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.html (hunk 1 above)
Comment (ycheng627) [reply diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.html]: Deleted.

File: diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts (hunk 3 above)
Comment (ycheng627) [reply diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts]: Changed to hasPassed = false.

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts (hunk 2 above)
Comment (ycheng627) [reply diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts]: Done.

File: diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts (hunk 5 above)
Comment (ycheng627) [reply diagnostics-app/src/app/rma/test-viewer/routine-v2-test/routine-v2-test.component.ts]: Done.

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.css [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+.active {
+  font-weight: 600;
+  color: var(--app-sidenav-item-active-color);
+}
+
+.mat-subheader {
+  color: var(--app-sidenav-subheader-color);
+  font-size: 20px;
+  /* background-color: red; */
```
[the remark this answers is not in the document]
Comment (ycheng627) [reply diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.css:13]: Done.

File: diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+  stopTests() {
+    this.testOrchestrator.stopTests();
+  }
+
+  getCurrentTestIndex$() {
+    return this.testOrchestrator.getCurrentTestIndex$();
+  }
+
+  hasPassed(index: number): 'pending' | 'passed' | 'failed' {
```
Comment (andy920262) [inline diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts:61]: `hasPassed` sounds like it should return a boolean.  Prefer something like `getTestResult` or `getTestStatus`

Comment (ycheng627) [reply diagnostics-app/src/app/rma/rma-sidebar/rma-sidebar.component.ts]: Done.

**Pull request merged**